src/solvers/file_solution_history.C

```diff
@@ -284,8 +181,13 @@ void FileSolutionHistory::retrieve(bool is_adjoint_solve, Real time)
   }
   else
   {
-    // Read in the primal solution stored at the current recovery time from the disk
-    _system.get_equation_systems().read (stored_sols->second, READ, EquationSystems::READ_DATA | EquationSystems::READ_ADDITIONAL_DATA);
+    // // If we are not in the adjoint loop, we could be in the primal loop again having solved
+    // // only the primal problem, or in a primal postprocessing stage to evaluate a QoI for example.
+    // // If we can find an adjoint solution file, read that in, else read in the primal file
+    if(dynamic_cast<FileHistoryData &>(*(stored_datum->second)).get_adjoint_filename().empty())
```

Comment:
Why are we relying on this rather than is_adjoint_solve?

Reply:
Thanks for catching this. Will fix.

Reply:
> See lines 510 to 526

From IMs, copying it here for the record.

Yeah, this makes sense now.  In hindsight we really need to rework this API, but this PR isn't creating any new deficiencies, just working around the original flaws ... and I'm not sure what the right long-term fix is yet, either.